src/instructlab/model/evaluate.py

```diff
@@ -349,6 +359,7 @@ def evaluate(
     few_shots,
     batch_size,
     sdg_path,
+    gpus,
```

Comment:
I don't see `gpus` being set via the config anywhere in `evaluate.py`. Is that going to be follow up work? Are we expecting the `--gpus` flag to just be set to the config somehow if it's not overridden?

Reply:
The logic currently takes the values in order or precedence from --gpus, config, accept default from serving (which would be 1).  Not quite following where else it should be set.

Reply:
Apologies, my previous comment wasn't clear.

Where in `evaluate.py` is the value being set from config? I only see the value from the flag being propagated down into `launch_server()`.

Also what would happen if `--tensor-parallel-size` was already in the list of `vllm-args` in the `serve` section of `config.yaml`? Right now it seems `--tensor-parallel-size` would be in `vllm-args` twice.

Reply:
click looks up defaults from the context's `default_map` attribute. The default map is configured in `instructlab.configuration` and `instructlab.model.model`. The `evalulate` subcommand uses `ctx.default_map["evalulate"]` as base map.

My PR #1591 tries to make it easier to understand and less messy.

Reply:
> Also what would happen if `--tensor-parallel-size` was already in the list of `vllm-args` in the `serve` section of `config.yaml`?

Updated to not override.  I am hoping we add --gpus to serve as well.  If we do, this check can be replaced by just passing gpus to serve.